hnix-store-remote/src/System/Nix/Store/Remote/Protocol.hs

```diff
@@ -168,17 +171,25 @@ runStore = runStoreOpts defaultSockPath "/nix/store"
 
 runStoreOpts
   :: FilePath -> FilePath -> MonadStore a -> IO (Either String a, [Logger])
-runStoreOpts sockPath storeRootDir code = do
-  bracket (open sockPath) (Network.Socket.close . storeSocket) run
+runStoreOpts path = runStoreOptsInner S.AF_UNIX (SockAddrUnix path)
+
+runStoreOptsTCP
+  :: String -> Int -> FilePath -> MonadStore a -> IO (Either String a, [Logger])
+runStoreOptsTCP host port storeRootDir code = do
+  let hints = S.defaultHints
```

Comment:
No point in creating additional abstraction here: https://wiki.haskell.org/Fairbairn_threshold